### PR TITLE
[VITS] Fix init test

### PR DESCRIPTION
**Pull request by sanchit-gandhi**
# What does this PR do?

Fixes https://github.com/huggingface/transformers/pull/24085#issuecomment-1704897914. In short, the VITS initialisation test was flaky: some of the parameters initialised with uniform values would exceed the initialiser range `[-1, 1]`. 

These violating parameters would always be in the latter stages of the HiFi GAN vocoder, so we can assume this was due to an accumulation of numerical errors.

This PR reduces the size of the HiFi GAN vocoder by a factor of 2, negating the accumulation of these errors. The test now passes over 20 iterations, but we should watch out if it turns out flaky over a larger range.

### CODE REVIEW

File: tests/models/vits/test_modeling_vits.py [hunk 1]
```diff
@@ -135,7 +147,7 @@ def create_and_check_model_forward(self, config, inputs_dict):
         attention_mask = inputs_dict["attention_mask"]
 
         result = model(input_ids, attention_mask=attention_mask)
-        self.parent.assertEqual(result.waveform.shape, (self.batch_size, 11008))
+        self.parent.assertEqual((self.batch_size, 624), result.waveform.shape)
```
Comment (sanchit-gandhi) [inline tests/models/vits/test_modeling_vits.py:150]: Output dims change after modifying the HiFi GAN vocoder

**Review by ydshieh — APPROVED** [review]
Works for me.

I ran it 100 times and got 8 failures.

Maybe add something like

```python
@is_flaky(max_attempts=3, description="...(some comment)...")
```

**Comment by ylacombe** [comment]
Hi @sanchit-gandhi , I'm not sure how accumulation of numerical errors could appear during initialization, could you expand on this a bit?

Also, would it be possible that the error comes from the [`init_weights`](https://github.com/huggingface/transformers/blob/ab8cba824e3887d90cb9f4d5866fde9243f2c9fe/src/transformers/models/vits/modeling_vits.py#L1284-L1288)'s lack of `ConvTranspose1d` initialization ?

**Review by amyeroberts — APPROVED** [review]
Thanks for fixing!

**Comment by sanchit-gandhi** [comment]
The failing module is actually a vanilla conv1d layer (rather than a conv transpose): https://github.com/huggingface/transformers/blob/604a6c51ae0b4ce5e8213ea86ed9c71373223a5d/src/transformers/models/vits/modeling_vits.py#L449

I haven't looked too deep into it, but I presumed the error was occurring due to using the [kaiming normal intialiser](https://paperswithcode.com/method/he-initialization) for the conv layers: 
https://github.com/huggingface/transformers/blob/604a6c51ae0b4ce5e8213ea86ed9c71373223a5d/src/transformers/models/vits/modeling_vits.py#L1285

Feel free to dive into it more if you want to find the source error! But based on the values we're getting it just looks like an instance of flakiness (the mean weights are 1.03 instead of 1.0)

File: tests/models/vits/test_modeling_vits.py [hunk 2]
```diff
@@ -168,30 +181,30 @@ def test_model_forward(self):
     def test_determinism(self):
         pass
 
-    # TODO: Fix me (ydshieh)
-    @unittest.skip("currently failing")
+    @is_flaky(description="Weight initialisation for the VITS conv layers sometimes exceeds the kaiming normal range")
```
Comment (ydshieh) [inline tests/models/vits/test_modeling_vits.py:184]: let's put `max_attempts=3` here. The default is `5`. Not a big deal but just my preference.

Comment (sanchit-gandhi) [reply tests/models/vits/test_modeling_vits.py]: Good for me! Resolved in https://github.com/huggingface/transformers/pull/25945/commits/4f970d273b390c729be07b435134af8f90e4ba01

**Review by ydshieh — APPROVED** [review]

**Pull request merged**